tests/CppWinRTAuthoringTests.cpp

```diff
@@ -0,0 +1,208 @@
+    REQUIRE(prop2 == value2);
+    REQUIRE(prop2() == value2);
+    REQUIRE(prop2 == prop2());
+    REQUIRE(prop2 == prop2);
+
+    wil::single_threaded_rw_property<std::string> prop3("foo");
+    REQUIRE(prop3 == "foo");
+    REQUIRE(prop3() == "foo");
+    REQUIRE(prop3.length() == 3);
+    prop3 = "bar";
+    REQUIRE(prop3 == "bar");
+    auto& prop3alias = prop3("baz");
+    REQUIRE(prop3 == "baz");
+    prop3alias = "foo";
+    REQUIRE(prop3 == "foo");
+}
+
+TEST_CASE("CppWinRTAuthoringTests::ReadWriteFromReadOnly", "[property]")
+{
+    int value = 42;
+    wil::single_threaded_property<int> prop(value);
+    REQUIRE(prop == value);
+    REQUIRE(prop() == value);
+    REQUIRE(prop == prop());
+    REQUIRE(prop == prop);
+
+    wil::single_threaded_rw_property<int> prop2(prop);
+    REQUIRE(prop2 == value);
+    REQUIRE(prop2() == value);
+    REQUIRE(prop2 == prop());
+    REQUIRE(prop2 == prop);
+
+    int value2 = 43;
+    prop2 = value2;
+    REQUIRE(prop2 == value2);
+    REQUIRE(prop2() == value2);
+    REQUIRE(prop2 == prop2());
+    REQUIRE(prop2 == prop2);
+
+    wil::single_threaded_rw_property<int> prop3{ prop };
+    REQUIRE(prop3 == value);
+    REQUIRE(prop3() == value);
+    REQUIRE(prop3 == prop());
+    REQUIRE(prop3 == prop);
+
+    wil::single_threaded_rw_property<int> prop4 = prop;
+    REQUIRE(prop4 == value);
+    REQUIRE(prop4() == value);
+    REQUIRE(prop4 == prop());
+    REQUIRE(prop4 == prop);
+}
+
+TEST_CASE("CppWinRTAuthoringTests::InStruct", "[property]")
+{
+    struct TestStruct
+    {
+        wil::single_threaded_property<int> Prop1{ 42 };
+        wil::single_threaded_rw_property<int> Prop2{ 1 };
+        wil::single_threaded_property<int> Prop3{ 44 };
+        void foo()
+        {
+            Prop1 = -42;
+        }
+    };
+
+    TestStruct test;
+    test.Prop2 = 43;
+
+    REQUIRE(test.Prop1 == 42);
+    REQUIRE(test.Prop2 == 43);
+    REQUIRE(test.Prop3 == 44);
+
+    test.Prop2 = 45;
+    REQUIRE(test.Prop2 == 45);
+
+    REQUIRE(test.Prop1() == 42);
+    test.Prop2(99);
+    REQUIRE(test.Prop2() == 99);
+    test.Prop2(22)(33);
+    REQUIRE(test.Prop2() == 33);
+}
+
+#ifdef WINRT_Windows_Foundation_H
+TEST_CASE("CppWinRTAuthoringTests::Events", "[property]")
+{
+    struct Test
+    {
+        wil::simple_event<int> MyEvent;
+
+        wil::typed_event<winrt::Windows::Foundation::IInspectable, int> MyTypedEvent;
+    } test;
+
+    auto token = test.MyEvent([](winrt::Windows::Foundation::IInspectable, int args) { REQUIRE(args == 42); });
+    test.MyEvent.invoke(nullptr, 42);
+    test.MyEvent(token);
+
+    auto token2 = test.MyTypedEvent([](winrt::Windows::Foundation::IInspectable, int args) { REQUIRE(args == 42); });
+    test.MyTypedEvent.invoke(nullptr, 42);
+    test.MyTypedEvent(token2);
+}
+#endif // WINRT_Windows_Foundation_H
+
+// if using MSVC since we need the embedded manifest bits to use XAML islands
+#if _MSVC_LANG >= 201703L && defined(WINRT_Windows_UI_Xaml_Data_H)

```

Comment:
Note that this is _not_ checking for MSVC as `clang-cl` also provides a definition for `_MSVC_LANG`

Reply:
actually this shouldn't be needed now that I'm embedding the manifest via cmake in the first place